### DISCUSSION TITLE
Feature: auto-dismiss markers / intermediate destinations

### DISCUSSION THREAD

**Discussion by mwerle** [osmandapp/OsmAnd#18662] [Ideas]
I use OsmAnd as my primary means of navigation on a motorcycle.  I usually pre-plan routes and use Markers to specify intermediate destinations. This allows me to quickly reorganise and replan the route if I've decided to take a detour or OsmAnd has stopped the route for some reason and I have to replan it. Most of the time I set intermediate destinations purely to force the routing to go along a particular road rather than actually wanting to stop there.

Interacting with the device while riding a motorcycle is obviously highly discouraged, as well as difficult to do with gloves, so having screen popups requesting me to dismiss a marker etc is not feasible.

One rather annoying thing I've had occur regularly is that the intermediate destinations are not "passed" automatically, or at least, not always, which means that OsmAnd will continuously attempt to route me back to the missed intermediate destination even if I am hundreds of kilometers past it and about to pass the next one.

It would be great if I could specify a radius (globally as well as per marker/waypoint) in which the marker/waypoint is considered to have been passed. Sometimes I would want to go past the marker closely (eg, a viewpoint), other times it would be good enough to just generally go past (eg, marker is in a village, but I've decided to use the bypass instead).

This feature could work and/or be developed in conjunction with feature request #17372.

Thank you for your hard work on this awesome product; it has made motorbike touring in Japan much more feasible!

**Comment by pebogufi** [comment]
Try the following workaround: 
create a quick action "Delete next destination point" and activate it after passing a destination point which will not be removed automatically.
Such points are a problem because they will not be removed if they are  a bit away from the route and accidentially on a street nearby the route. This may happen when creating the destiation point at a low zoomlevel.

**Reply by mwerle** [reply]
Thank you for your suggestion (and apologies for my slow response) - but see my point above: interactive with a device while motorcycling is dangerous as well as impossible while wearing gloves.  This needs to be automatic.

Incidentally this is also why I've completely stopped using Google Maps for navigating on the motorcycle because of its incessant popups (alternative route suggestions, not dismissing reached waypoints, etc).  While OsmAnd thankfully does not have popups which must be manually dismissed, my big issue is it not recognising that I've passed a waypoint. It's not always feasible to stop (eg on a motorway) to muck about with the device in such a situation.

**Comment by muusik** [comment]
I would like to bump and piggyback on this conversation to suggest a new feature: an approximate intermediate destination. It's a circular area that the user gets routed through and automatically dismissed, and its size should approximate the user's fingertip at the zoom level at the time it gets added. In my estimation, this (along with the possibility to add IDs between specific other IDs, not just second, second-to-last or after-the-last) would bring alternate route planning up to the high standard of what OsmAnd is capable of.

I don't know enough about routing to suggest a good implementation for it, but here's a bad version that would be good enough for me: Start by adding a regular intermediate destination and calculating a route through it, then draw a "fingertip-sized" circle around it, then find the points where the route and the circle intersect, then replace the initial intermediate destination with both of these points, in the correct order (though, in principle, one may suffice), then recalculate the route again.

Well, come to think of it, simply replacing the intermediate destinations like that would interfere with further route modifications in the case that the route within one such circle were to change. To do it properly, the original point should be considered for the calculation of the entire route, and my initial bad algorithm applied locally to each approximate intermediate destination in a separate step. Or perhaps there's a better way to do it to begin with.

But if I can revise a route by going "I'd like to go through somewhere around here, though" and end up with one or two intermediate destinations that are close enough to where I fat-fingered the screen, that would be good enough. The feature can then be polished further by drawing a nice translucent circle on the map, hiding the extra two intermediate destinations behind one entity, etc, but these are nice-to-haves.
